app/services/replication_transfer_updater.rb

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) 2015 The Regents of the University of Michigan.
+# All Rights Reserved.
+# Licensed according to the terms of the Revised BSD License
+# See LICENSE.md for details.
+
+class ReplicationTransferUpdater
+
+  class << self
+    def update(record, params)
+      build_update(record, params).update
+    end
+
+    def build_update(record, params)
+      case
+      when CancelUpdate.matching_update?(record, params)
+        CancelUpdate.new(record, params)
+      when FixityValueUpdate.matching_update?(record, params)
+        FixityValueUpdate.new(record, params)
+      when StoreRequestedUpdate.matching_update?(record, params)
+        StoreRequestedUpdate.new(record, params)
+      when StoredUpdate.matching_update?(record, params)
+        StoredUpdate.new(record, params)
+      else
+        ReplicationTransferUpdate.new(record, params)
+      end
+    end
+
+  end
+
+
+  # A generic update
+  class ReplicationTransferUpdate
+
+    # @param [ReplicationTransfer] record
+    # @param [HashWithIndifferentAccess] params The params hash of the update
+    def initialize(record, params)
+      @record = record
+      @params = params
+    end
+    def update
+      record.update(params)
+    end
+    def we_requested?
+      record.from_node&.namespace == Rails.configuration.local_namespace
+    end
+    def we_replicating?
+      record.to_node&.namespace == Rails.configuration.local_namespace
+    end
+
+    private
+    attr_reader :record, :params
+  end
```

Comment:
Yea, consolidate the string case management as much as possible.  It's too easy to forget about it and run into obscure bugs just because the case doesn't match up.